tests/server/utils/data_callback.py

```diff
@@ -0,0 +1,70 @@
+import json
```

Comment:
I'd recommend renaming this file to `test_data_callback.py`, as otherwise IIRC it won't be picked up by `pytest`.

Reply:
Sure!